{
    "type": "vulnerability",
    "spec_version": "2.1",
    "id": "vulnerability--f80388de-da14-42a7-b2c7-aecb32d08dfa",
    "created_by_ref": "identity--748e6444-f073-4c50-b558-f49be8897a81",
    "created": "2021-08-25T01:15:00.000Z",
    "modified": "2021-08-30T19:29:00.000Z",
    "name": "CVE-2021-40083",
    "description": "Knot Resolver before 5.3.2 is prone to an assertion failure, triggerable by a remote attacker in an edge case (NSEC3 with too many iterations used for a positive wildcard proof).",
    "external_references": [
        {
            "source_name": "cve",
            "external_id": "CVE-2021-40083"
        },
        {
            "source_name": "cwe",
            "external_id": "CWE-617"
        },
        {
            "source_name": "https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1169",
            "url": "https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1169"
        },
        {
            "source_name": "cve2stix",
            "description": "This object was created using cve2stix from the Signals Corps.",
            "url": "https://github.com/signalscorps/cve2stix"
        }
    ],
    "object_marking_refs": [
        "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
    ],
    "extensions": {
        "extension-definition--b2b5f2cd-49e6-4091-a0e0-c0bb71543e23": {
            "extension_type": "property-extension",
            "impact": {
                "baseMetricV2": {
                    "acInsufInfo": false,
                    "cvssV2": {
                        "accessComplexity": "LOW",
                        "accessVector": "NETWORK",
                        "authentication": "NONE",
                        "availabilityImpact": "PARTIAL",
                        "baseScore": 5.0,
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "vectorString": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
                        "version": "2.0"
                    },
                    "exploitabilityScore": 10.0,
                    "impactScore": 2.9,
                    "obtainAllPrivilege": false,
                    "obtainOtherPrivilege": false,
                    "obtainUserPrivilege": false,
                    "severity": "MEDIUM",
                    "userInteractionRequired": false
                },
                "baseMetricV3": {
                    "cvssV3": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    },
                    "exploitabilityScore": 3.9,
                    "impactScore": 3.6
                }
            },
            "lastModifiedDate": "2021-08-30T19:29Z",
            "nvd_cve": {
                "CVE_data_meta": {
                    "ASSIGNER": "cve@mitre.org",
                    "ID": "CVE-2021-40083"
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "en",
                            "value": "Knot Resolver before 5.3.2 is prone to an assertion failure, triggerable by a remote attacker in an edge case (NSEC3 with too many iterations used for a positive wildcard proof)."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "en",
                                    "value": "CWE-617"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1169",
                            "url": "https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1169",
                            "refsource": "MISC",
                            "tags": [
                                "Patch",
                                "Third Party Advisory"
                            ]
                        }
                    ]
                }
            },
            "publishedDate": "2021-08-25T01:15Z"
        }
    }
}